agent/consul/internal_endpoint_test.go

```diff
@@ -175,6 +213,21 @@ func TestInternal_NodeDump(t *testing.T) {
 	if !foundFoo || !foundBar {
 		t.Fatalf("missing foo or bar")
 	}
+
+	require.Equal(t, 1, len(out2.ImportedDump))
+	require.Equal(t, "peer1", out2.ImportedDump[0].PeerName)
+	require.Equal(t, "foo-peer", out2.ImportedDump[0].Node)
+
+	// test filter on nodes
+	var out3 structs.IndexedNodeDump
+	req2 := structs.DCSpecificRequest{
+		Datacenter:   "dc1",
+		QueryOptions: structs.QueryOptions{Filter: "friend in PeerName"},
```

Comment:
Not too familiar with filters but it is possible to filter for not empty peerNames to only show peered nodes?

Reply:
@kisunji I'm not sure that would be useful here since there's already tests for filtering the `.Dump` field. 

The question we had was about whether the new `ImportedDump` could be filtered on.

Reply:
we can -- we can basically construct any boolean expression over the structs

https://www.consul.io/api-docs/features/filtering#creating-expressions

I can add a test for that right now